Mlem/Logic/Generic Merge.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  Generic Merge Sort.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2023-06-27.
+//
+
+import Foundation
+
+/**
+ Performs merge on two sorted arrays, returning the result.
+ 
+ The arrays must be sorted such that compare(a[0], a[1]) returns true
```

Comment:
Nit:

it could be worth adding an `assert` at the top of the function to test the above for the two inputs, would catch during development if we ever tried to send some arrays in that weren't already sorted?